### ISSUE TITLE
ability to resolve FQDN

### ISSUE THREAD

**Issue by kamilmirza** [ipinfo/cli#52]
is there any option to get ipinfo of a domain name?
suppose `ipinfo google.com`
would be very much useful with DDNS domains

**Comment by UmanShahzad** [comment]
Yeah we can implement that, but it would use the user's local DNS setup to resolve the domain, rather than for example make a request to ipinfo.io to do that.

**Comment by coderholic** [comment]
I'm not sure there's a compelling use case to build this in - you can just do it with this, which makes it clear where the lookup is happening:

```
$ dig +short google.com | ipinfo
{
  "142.250.217.78": {
    "ip": "142.250.217.78",
    "hostname": "sea09s29-in-f14.1e100.net",
    "city": "Seattle",
    "region": "Washington",
    "country": "US",
    "country_name": "United States",
    "loc": "47.6062,-122.3321",
    "org": "AS15169 Google LLC",
    "postal": "98111",
    "timezone": "America/Los_Angeles"
  }
}
```

**Comment by UmanShahzad** [comment]
Yeah the part about not knowing for sure, or not being able to control fully, where the IP of the domain is retrieved from kind of reduces the value of this: DNS servers will for example give different IPs via round-robin, geolocation, caching, and possibly more context. If we were to do the DNS lookup on the servers, for example, the results will usually be very different than if done on the client.

**Comment by UmanShahzad** [comment]
We can maybe add a simple version of this under a subcommand `fqdnip` (FQDN IP, similar to `myip` "My IP"), e.g. `ipinfo fqdnip google.com`. We would just do the DNS lookup and hand it over to the work that `ipinfo <ip>` is doing.

**Comment by eacp** [comment]
> I'm not sure there's a compelling use case to build this in - you can just do it with this, which makes it clear where the lookup is happening:
> 
> ```
> $ dig +short google.com | ipinfo
> {
>   "142.250.217.78": {
>     "ip": "142.250.217.78",
>     "hostname": "sea09s29-in-f14.1e100.net",
>     "city": "Seattle",
>     "region": "Washington",
>     "country": "US",
>     "country_name": "United States",
>     "loc": "47.6062,-122.3321",
>     "org": "AS15169 Google LLC",
>     "postal": "98111",
>     "timezone": "America/Los_Angeles"
>   }
> }
> ```

The dig command is not part of windows.

**Comment by eacp** [comment]
> We can maybe add a simple version of this under a subcommand `fqdnip` (FQDN IP, similar to `myip` "My IP"), e.g. `ipinfo fqdnip google.com`. We would just do the DNS lookup and hand it over to the work that `ipinfo <ip>` is doing.

I can make a pull req for that. Or maybe insert such behaviour as a special case in the normal ipinfo command

**Comment by coderholic** [comment]
> The dig command is not part of windows.

You can get dig for windows: https://docs.digitalocean.com/products/networking/dns/resources/use-dig/

I think nslookup comes bundled though

**Comment by UmanShahzad** [comment]
I guess if you made a PR I would be open to accepting it, but we should add a disclaimer in the subcommand description (in its `-h` message) that the IP looked up may not be the IP others around the world see for the same domain, that it may be cached, etc.

**Comment by eacp** [comment]
@UmanShahzad another important consideration is that dns servers might give different IP addresses depending on the geographic location for distributed applications, like CDNs. Still, the CLI could emulate the behaviour of commands such as `ping` or `traceroute`, which automatically solve the ip address when using domain names.

**Comment by UmanShahzad** [comment]
> dns servers might give different IP addresses depending on the geographic location for distributed applications

Yes this is the point I had in mind when mentioning that others around the world may not see the same IP  for the same domain, because some globally distributed dns servers will return the IP of the server closest to your location.

**Comment by eacp** [comment]
Some sort of warning could be displayed if neccesary.

**Comment by UmanShahzad** [comment]
> Some sort of warning could be displayed if neccesary.

I recommended:

> we should add a disclaimer in the subcommand description

**Comment by eacp** [comment]
I agree with such reccomendation

**Comment by UmanShahzad** [comment]
@eacp Definitely a subcommand as `fqdnip`, not `domain`. But I'll handle this one, no worries.

**Comment by kamilmirza** [comment]
or it can be `fqdninfo`

**Comment by UmanShahzad** [comment]
`fqdninfo` or `domain` both sound like they're giving domain-specific information, which is something IPinfo provides. But this command isn't doing that so such names aren't the best choice.

**Comment by eacp** [comment]
@UmanShahzad  IMHO, it would be more intuitive if the CLI replicated the behaviour of ping and traceroute: use the ip as given by the user, and IF the user entered a FQDN, solve the IP address and use that IP address.

```
$ ping github.com

Haciendo ping a github.com [140.82.113.4] con 32 bytes de datos:
Respuesta desde 140.82.113.4: bytes=32 tiempo=149ms TTL=52
Respuesta desde 140.82.113.4: bytes=32 tiempo=79ms TTL=52
Respuesta desde 140.82.113.4: bytes=32 tiempo=77ms TTL=52
Respuesta desde 140.82.113.4: bytes=32 tiempo=77ms TTL=52
```

Notice how I was allowed to just type `github.com` instead of an IP address.

**Comment by coderholic** [comment]
@eacp that's an interesting idea. I think I actually prefer the idea of supporting `ipinfo $domain` and returning the details for the IP that domain resolves to, rather than having a separate ip resolution command, which feels out of scope for the cli.

**Comment by UmanShahzad** [comment]
Alright we've finalized on going with the `ipinfo $domain` model. I'll implement this soon and include it in https://github.com/ipinfo/cli/pull/59.

Thanks for all your inputs!
